test/hack/bin/fetch-tools.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# let's have reproducable tests - pin to specific versions
+ARCHITECT_VERSION="6.5.0"
+PROMETHEUS_VERSION="2.36.2"
```

Comment:
I would rather use the current Prometheus version we are running

```suggestion
PROMETHEUS_VERSION="2.28.1"
```